### PR TITLE
Switch cpld_bmark_p8 test to rectilinear 1/2deg grid; add Mesh creation info to wiki; Update P8 sfc ICs for FV3; Update WW3_inputdata directory; Bug fix for #1309: false reports of test failures

### LINKED ISSUE #1203: Update ICs for P8c surface fields
## Description
The surface fields for p8c ICs were updated.  While likely sufficient for regression testing for the existing ICs, wanted to make everyone aware that there is a change in what we're running for the prototypes, so we should probably update these inputs when we get a chance.  

Essentially we need to do the following: 

for CDATE  in  2012010100  2012040100	2012070100  2012100100	2013010100  2013040100	2013070100  2013100100

cp  /scratch2/NCEPDEV/climate/climpara/S2S/IC/GEFS-NoahMP-aerosols-p8c/${CDATE}/gfs/C384/INPUT/sfc*.nc  /scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/BM_IC-20220207/${CDATE}/gfs_p8c/C384_L127/INPUT/

### LINKED ISSUE #1297: Switch bmark_p8 test to a rectilinear 1/2deg grid
## Description
The 1/4 deg tripole grid is too slow for P8 to run w/in the 8-hour time frame; the corresponding regression test needs to be switched to the 1/2 deg rectilinear grid that will be used in the P8 runs.

## Solution

Update the RT scripts to use a mesh and wave domain different than the ocean and ice.

### LINKED ISSUE #1309: Regression tests falsely marked as failed when prior failed attempt finishes after next successful attempt starts
## Description

Sometimes the regression test system will falsely mark a test as failed if it was resubmitted. This is due to a race condition described below. The easiest solution is for the jobs to delete their `fail_(number)` files at the end of the job, not just the beginning.

### How the System is Designed

The regression test system makes `fail_*` files to indicate a regression test failed. The intended order of operations in the job is this:

1. First job starts.
2. First job deletes the `fail_(number)` file
3. First job runs the test, and the test fails.
4. First job makes the `fail_(number)` file to indicate the test failed.
5. First job exits.
6. Job is resubmitted.
7. Second job deletes the `fail_(number)` file
8. Second job runs the test
9. Second job's test succeeded, and the job exits

### What can go Wrong

The trouble is, this can happen in a different order if the first job is killed or preempted. In that situation, a SIGINT handler makes the fail file, and that can happen after the job completed. 

1. First job starts.
2. First job deletes the `fail_(number)` file
3. First job runs the test, but the test is interrupted abnormally
4. First job is killed
5. Job is resubmitted
6. Second job starts.
7. Second job deletes the `fail_(number)` file
8. First job's SIGINT handler creates the `fail_(number)` file
9. First job exits.
10. Second job runs the test
11. The test succeeded, and the job exits
12. The `fail_(number)` file is still present because the second job did not delete it.

### The Solution

The simplest solution is for the job to delete the `fail_(number)` file at the end if the test succeeded. Then the order of operations is:

1. First job starts.
2. First job deletes the `fail_(number)` file
3. First job runs the test, but the test is interrupted abnormally
4. First job is killed
5. Job is resubmitted
6. Second job starts.
7. Second job deletes the `fail_(number)` file
8. First job's SIGINT handler creates the `fail_(number)` file
9. First job exits.
10. Second job runs the test
11. Second job's test succeeded
12. Second job deletes the `fail_(number)` file.
13. The regression test system considers the test successful because the `fail_(number)` file is absent.

### To Reproduce:
This is frequently a problem on Jet, but other systems can have it too.
1. Run the regression test system many times.
2. Witness the problem.

### PR DESCRIPTION
**Pull request by DeniseWorthen**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [x] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsibility to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

- Changes domain for the cpld_bmark_p8 test to use the gwes_30m domain.
- Updates WW3 input data for new mod_def files created using current intel 2022 compiler. 
-  A new ``WW3_input_data_20220624`` directory has been added to hera and will need to be sync'd across all platforms. The updated mod_def files will change baselines for all tests using the wave model.
- The ``BM_IC-20220207`` directory on hera has a new ``p8c`` directory for each date; this needs to be sync'd across all platforms.
- Adds information on mesh creation to wiki.  

### Issue(s) addressed

- fixes #1297 

## Testing

The full rt.conf was run on hera.intel at ba1b2a63 using a copy of the current ``input-data-20220414/WW3_input_data_20220418/`` with new ``mesh.gwes_30m.nc`` and ``mod_def.gwes_30m`` added. All tests passed with the exception of ``cpld_bmark_p8``, which ran to completion.

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss2.intel
- NA wcoss_cray
- NAA wcoss_dell_p3
- [x] opnReqTest for newly added/changed feature. oRTs for cpld_bmark_p8 (std_base,thr,rst,dcp) passed at 7e4a9ba 
- [ ] CI

## Dependencies

None. Script level changes only.

### CODE REVIEW

**Comment by junwang-noaa** [comment]
@JessicaMeixner-NOAA May I ask what is the impact of Intel compiler 2022 to ww3 mod_def input files? Will mod_def files need to be regenerated every time when we update the compiler version? Thanks

**Comment by JessicaMeixner-NOAA** [comment]
@junwang-noaa the binary mod_def files change based on compiler.  While they do not strictly need to be remade when a new version of the compiler is used (unless you are changing endianness), when you later update the mod_defs it's not unreasonable to expect an answer change.  There are some pre-calculations that happen and are stored in the mod_def files, so just like how its not unexpected for the model answers to change with a change in compiler, it's not unexpected for the mod_defs to change.

**Comment by junwang-noaa** [comment]
@JessicaMeixner-NOAA Thanks for the information. We do expect results change when compiler is updated. In general we don't expect endianness change because of compiler version. Is that correct the pre-computed data in mod_def can't be used when a new compiler version is used? It looks to me it's a dependency of input data on compiler, which is required by ww3 only. May I ask if you can share the computation code used to generate the results in mod_def? Thanks.

**Comment by DeniseWorthen** [comment]
@BinLiu-NOAA Following this discussion, I believe we should also have a new wind input field for the regional HAFSW RT created using the new intel 2022 compiler.

**Comment by junwang-noaa** [comment]
> @JessicaMeixner-NOAA Thanks for the information. We do expect results change when compiler is updated. In general we don't expect endianness change because of compiler version. Is that correct the pre-computed data in mod_def can't be used when a new compiler version is used? It looks to me it's a dependency of input data on compiler, which is required by ww3 only. May I ask if you can share the computation code used to generate the results in mod_def? Thanks.

@SMoorthi-emc FYI. I am still trying to understand what dependencies of  the mod_def files on compiler version.

**Comment by jkbk2004** [comment]
@DeniseWorthen Should I start rsync from hera?

**Comment by DeniseWorthen** [comment]
@jkbk2004 Yes, you can sync both the new p8c and WW3_input_data_20220624 directories. 


@BrianCurtis-NOAA I know we're not running coupled on wcoss2 yet, but I assume the input directories are present and need to be updated?

**Comment by BrianCurtis-NOAA** [comment]
> @BrianCurtis-NOAA I know we're not running coupled on wcoss2 yet, but I assume the input directories are present and need to be updated?

Yes, they are there and will need updated.

**Comment by jkbk2004** [comment]
> @jkbk2004 Yes, you can sync both the new p8c and WW3_input_data_20220624 directories.
> 
> @BrianCurtis-NOAA I know we're not running coupled on wcoss2 yet, but I assume the input directories are present and need to be updated?
@DeniseWorthen p8c for all files 2012 and 2013, right?

**Comment by DeniseWorthen** [comment]
@jkbk2004 Yes, the ``BM_IC-20220207`` directory will need new p8c files for all the dates (8 dates total).

**Comment by jkbk2004** [comment]
@SamuelTrahanNOAA @DeniseWorthen can we combine #1312 to this PR? so that we can test out especially jet issue today.

**Comment by DeniseWorthen** [comment]
That is fine w/ me. We're both up-to-date so I will merge in PR #1312.

**Comment by jkbk2004** [comment]
@SamuelTrahanNOAA do you have access to cheyenne? I am not sure the compile failure report comes from the rt.sh change?

**Comment by jkbk2004** [comment]
> @SamuelTrahanNOAA do you have access to cheyenne? I am not sure the compile failure report comes from the rt.sh change?
/glade/scratch/jongkim/UFS-RT-tests/rt-1292-intel/tests

**Comment by jkbk2004** [comment]
additional gnu compile failure cases on cheyenne at /glade/scratch/jongkim/UFS-RT-tests/rt-1292-gnu/tests

**Comment by SamuelTrahanNOAA** [comment]
The run_compile.sh is removing the wrong file. Hence, when the compilation fails and then succeeds, it is not removing the right failure file. I cannot commit to this branch, so somebody else will have to edit it.

In run_compile.sh, this:

```
    if [[ ${OPNREQ_TEST} == true ]] ; then
        rm -f $PATHRT/fail_opnreq_test_${COMPILE_NR}
    else
        rm -f $PATHRT/fail_test_${COMPILE_NR}
    fi
```

should be:

```
    if [[ ${OPNREQ_TEST} == true ]] ; then
        rm -f $PATHRT/fail_opnreq_compile_${COMPILE_NR}
    else
        rm -f $PATHRT/fail_compile_${COMPILE_NR}
    fi
```

Note the change from `_test_` to `_compile_`

**Comment by SamuelTrahanNOAA** [comment]
I fixed the bug in my PR, so if someone re-merges my branch into this branch, they should get the fix.

**Comment by jkbk2004** [comment]
@DeniseWorthen can you merge in Sam's latest fix? Or I can directly add the fix.

**Comment by DeniseWorthen** [comment]
Yes, I will add the fix from @SamuelTrahanNOAA

**Comment by jkbk2004** [comment]
> Yes, I will add the fix from @SamuelTrahanNOAA

Sure! let me test again.

**Comment by jkbk2004** [comment]
@DeniseWorthen cheyenne.intel BL creation one case atmaero_control_p8_rad crashes with mpt/mpi error: /glade/scratch/jongkim/rt-1292-intel/jongkim/FV3_RT/rt_15184/atmaero_control_p8_rad.  I will try to re-submit. Hera que is very slow today.

**Comment by jkbk2004** [comment]
> @DeniseWorthen cheyenne.intel BL creation one case atmaero_control_p8_rad crashes with mpt/mpi error: /glade/scratch/jongkim/rt-1292-intel/jongkim/FV3_RT/rt_15184/atmaero_control_p8_rad. I will try to re-submit. Hera que is very slow today.

atmaero_control_p8_rad runs ok on Cheyenne.intel.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: BL
[BL] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/978684257/20220706171519/ufs-weather-model
[BL] Error: Test datm_cdeps_mx025_gefs 100 failed in run_test failed 
[BL] Error: Test control_atmwav 104 failed in run_test failed 
Please make changes and add the following label back: orion-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: BL
[BL] Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/978684257/20220706230009/ufs-weather-model
Please make changes and add the following label back: jet-intel-BL

**Comment by DeniseWorthen** [comment]
Both orion failures seem to be system glitches. 

The atmwav test had 
```
108: Error in system call pthread_mutex_destroy: Device or resource busy
108:     ../../src/mpi/init/init_thread_cs.c:66
108: Abort(1090703) on node 0 (rank 0 in comm 0): Fatal error in PMPI_Init_thread: Other MPI error, error stack:
```

The datm test had  
```
147: [Orion-25-40:425553:0:425553] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
....
 26: [Orion-01-43:108691:0:108691]     address.c:1052 Assertion `*addr_version == UCP_OBJECT_VERSION_V2' failed: addr version 3
```
and then segfault.

**Comment by jkbk2004** [comment]
Yeah, I will work on orion. Jet has ecflow server connection issue. I am sending a help desk ticket.

File: tests/fv3_conf/cpld_control_run.IN [hunk 1]
```diff
@@ -45,10 +45,13 @@ cp    @[INPUTDATA_ROOT]/CICE_FIX/@[OCNRES]/mesh.mx@[OCNRES].nc .
 # WW3 fix/input
 if [[ $CPLWAV == .true. ]]; then
   cp    @[INPUTDATA_ROOT_WW3]/mod_def.points .
-  if [[ $OCNRES == 025 ]]; then
-     cp    @[INPUTDATA_ROOT_WW3]/mod_def.mx@[OCNRES]lite mod_def.ww3
+  if [[ $WAVDOMAIN == mx025 ]]; then
+     cp  @[INPUTDATA_ROOT_WW3]/mod_def.@[WAVDOMAIN]lite mod_def.ww3
```
Comment (junwang-noaa) [inline tests/fv3_conf/cpld_control_run.IN:49]: So the quarter degree coupled run is using a different mod_def file name "lite", will we consider use the same lite mod def files for other resolutions?

**Review by junwang-noaa — APPROVED** [review]

File: tests/fv3_conf/cpld_control_run.IN (hunk 1 above)
Comment (DeniseWorthen) [reply tests/fv3_conf/cpld_control_run.IN:49]: The "lite"  uses ww3 timesteps which violate the CFL criterion. The other domains do not need this special setting.

**Comment by SamuelTrahanNOAA** [comment]
Are my regression test system changes working? I cannot see if your workflows are resubmitting jobs, so I don't know.

**Comment by binli2337** [comment]
@DeniseWorthen For the cpld_bmark_p8 test, do you plan to revise the TEST_DESCR parameter to indicate that the resolution of wave model is rectilinear 30m or 1/2-degree?

**Comment by DeniseWorthen** [comment]
@binli2337 I will make this change in a later PR.

**Comment by jkbk2004** [comment]
> Are my regression test system changes working? I cannot see if your workflows are resubmitting jobs, so I don't know.

I think its working better on jet. It was rare case jet went thru by itself overnight. But currently ecflow issue on jet. I am using rocoto but it improves, I think.

**Review by JessicaMeixner-NOAA — APPROVED** [review]

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Pull request merged**